crypto/cmake/crypto-config.cmake

```
@@ -2,11 +2,22 @@ if(WIN32 OR UNIX OR APPLE)
     find_package(Threads REQUIRED)
 endif()
 
+# Allow static or shared lib to be used.
+# If both are installed, choose based on BUILD_SHARED_LIBS.
 if (BUILD_SHARED_LIBS)
```

Another way I could think of doing this is adding an extra "force"/"override" condition:
```suggestion
if (BUILD_SHARED_LIBS OR FORCE_USING_SHARED_LIBS)
```

But, you prefer it being implicit?

---

The existing logic prevents a user from building their own library as static when `libcrypto.so` is preinstalled from a package manager (see ticket in PR description).

With my change, whichever version of libcrypto is installed, that's what gets picked. That's what users expect.

But this does introduce ambiguity when both `libcrypto.so` and `libcrypto.a` are installed, so in that case I'm keeping the old behavior where BUILD_SHARED_LIBS drives the decision, to maintain backwards compatibility with how things were before.

I'm hesitant to add an explicit option. I spent a long time trying things out but I can see downsides to all approaches. We can add an explicit option later when we've really thought through all the issues. Nothing I'm doing here prevents that

---

Makes sense.